### ISSUE TITLE
Network Information Received

### ISSUE THREAD

**Issue by hypodyne** [IrcDotNet/IrcDotNet#72]
So you connect to a network(Freenode), and it sends back the network information.
251, 252, 253, 254, 255, 265, 266, 250

IrcDotNet throws the NetworkInformationReceived event with the details in e.Comment.

e.Comment sometimes contains a properly formatted string 

"There are 109 users and 84695 invisible on 32 servers"

and sometimes e.comment contains a number as a string.

"5"

The event is thrown 5 times out of the 8 responses. The last 3 never throw the event. 

The way the Network Information is structured, I can't use sender to get the information, because there is no way to tell which piece of information threw the event.

**Comment by wix-andriusb** [comment]
You want to tell me that it splits on 84695 between 8469 and 5 ?

**Comment by hypodyne** [comment]
I have a wireshark capture, and it seems if the response from the server is incorrect, then IrcDotNet also gets it wrong. Let me post a screenshot.

**Comment by hypodyne** [comment]
![wireshark](https://user-images.githubusercontent.com/28817500/52174801-4f63c780-27d4-11e9-8c56-6c9ef8b80e52.png)

Since freenode is sending packets which I presume are longer than the 1024 limit, it gets split in the wrong place. (in wireshark) and IrcDotNet can't handle that either. What happens is 265, 266, 250 don't throw an event. thats item 1.

**Comment by hypodyne** [comment]
![wireshark2](https://user-images.githubusercontent.com/28817500/52174852-2db71000-27d5-11e9-950d-7a12421b9cf0.png)

Another issue is that IrcDotNet throws the NetworkInformationReceived event with just 35 as the comment. What it should be is "35 :IRC Operators online". Without that extra info there is no telling which of the 7 different network information properties it comes from.

**Comment by txdv** [comment]
So the issues here are:

* If the server sends a response with > 1024 bytes, it gets split in the wrong place.
* Event 252 (RPL_LUSEROP) has an integer field and some additional info

Is that correct?

Here is a nice list of events https://www.alien.net.au/irc/irc2numerics.html

**Comment by hypodyne** [comment]
Yes, that is what I think is happening. 

A few network information events return an int (and that is all they return) in e.Comment. 

Really all events should be returning enough information so that you can post the response.

**Comment by hypodyne** [comment]
Yes here is one too [http://www.networksorcery.com/enp/protocol/irc.htm](http://www.networksorcery.com/enp/protocol/irc.htm) 

Notice 252

**Comment by hypodyne** [comment]
16

**Comment by hypodyne** [comment]
34

**Comment by hypodyne** [comment]
73

**Comment by hypodyne** [comment]
16, 34, 73 What do they mean. Without context they are useless.

**Comment by txdv** [comment]
This is not IRC, try to add all comments in one response :D
Also do you know what all of the meanings of the integer parameter in notice 252 are? Maybe there is a source which describes them? 

I think you are right that events in the library should contain everything that is needed, its just that in open source usually something happens when someone needs it or makes an issue out of what he dislikes about a certain library.

**Comment by hypodyne** [comment]
Just letting you guys know. It's not important at all.
